### PR TITLE
Home Button Fix

### PR DESCRIPTION
**Pull request by Tohaker**
This should solve #1036

### CODE REVIEW

**Comment by Tohaker** [comment]
I'm going to write some tests for this before merging, to make sure this doesn't happen again.

**Comment by schwabdidier** [comment]
@kuhlkrein can you check this PR ?

**Comment by kuhlkrein** [comment]
@Tohaker @schwabdidier I never used these mock tests before but it looks good to me

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java [hunk 1]
```diff
@@ -223,9 +222,11 @@ public HomeButton createHomeButtonInGameScreen(@NonNull GazePlay gazePlay, @NonN
         return homeButton;
     }
 
-    private void exitGame(@NonNull Stats stats, @NonNull GazePlay gazePlay, @NonNull GameLifeCycle currentGame) {
+    void exitGame(@NonNull Stats stats, @NonNull GazePlay gazePlay, @NonNull GameLifeCycle currentGame) {
 
-        pointersClear();
+        if (this.getConfiguration().isVideoRecordingEnabled()) {
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: Condition should be improved

We have 3 fields that are working together.
They are either all set or all unset.

Please try to
- put all 3 fields into a new class like VideoRecordingContext
- put VideoRecordingContext into GameContext
- if VideoRecordingContext is not null, clear it

Comment (Tohaker) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: Sounds good, I'll get on that later tonight

**Comment by schwabdidier** [comment]
I ask to sebastien to do it. I don't know when I can make the release, announcement etc if I don't do it today :(

**Comment by kuhlkrein** [comment]
@YannRobert , is it what you were talking about in your review ? Can we keep it like this or you want me to extract some functions from the VideoRecordingContext constructor ?

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/VideoRecordingContext.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        root.addEventFilter(MouseEvent.ANY, pointerEvent);
+        root.addEventFilter(GazeEvent.ANY, pointerEvent);
+    }
+
+    public void pointersClear() {
+        root.removeEventFilter(MouseEvent.ANY, pointerEvent);
+        root.removeEventFilter(GazeEvent.ANY, pointerEvent);
+        root.getChildren().remove(gazePointer);
+        root.getChildren().remove(mousePointer);
+        gameContext.getGazeDeviceManager().removeEventFilter(root);
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/VideoRecordingContext.java]: I guess this line should not be there

**Comment by Tohaker** [comment]
Why are there no unit tests for the new class?

**Comment by schwabdidier** [comment]
Sorry, it will be in next version. We try to release before Seb leaves the office, otherwise I only will have time next Monday...

**Pull request merged**